quantities/factories/SecondPower.cs

```diff
@@ -8,31 +8,31 @@
 
 namespace Quantities.Factories;
 
-public readonly struct SquareFactory<TQuantity, TCompound, TSquare, TLinear> : ISquareFactory<TQuantity, TSquare, TLinear>
-    where TLinear : Dimensions.IDimension, ILinear
-    where TSquare : ISquare<TLinear>, Dimensions.IDimension
-    where TCompound : ICompoundFactory<TQuantity, TLinear>, IAliasingCreate
+public readonly struct SecondPower<TCreate, TQuantity, TSquare, TLinear> : ISquareFactory<TQuantity, TSquare, TLinear>
+    where TCreate : struct, ICreate
     where TQuantity : IFactory<TQuantity>
+    where TSquare : ISquare<TLinear>, Dimensions.IDimension
+    where TLinear : Dimensions.IDimension, ILinear
 {
-    private readonly TCompound squareFactory;
-    public TCompound Square => this.squareFactory;
-    internal SquareFactory(in TCompound compound) => this.squareFactory = compound;
+    private readonly TCreate create;
+    public Compound<TCreate, TQuantity, TLinear> Square => new(in this.create, AllocationFree<PowerInjector<TCreate, Square>>.Item);
```

Comment:
If you renamed the factory, shouldn't you also rename the power? So it would look like: `new(in this.create, AllocationFree<PowerInjector<TCreate, RaiseToSecond>>.Item)` ? I'm not sure about the name though 😅

Reply:
I totally get your point. The naming here is very nifty.

First I'd like to keep the names of the `Measure.IDimension` implementations as is (i.e. `Linear`, `Square` and `Cubic`).

Then therese the matter of naming the factories. I sometimes get annoyed by naming a lot of things {Something}Factory or {Something}Builder or {Something}Exception. Mostly because the interface or base class already tells us if it's a factory, builder or exception. Why repeat that in the name?

So I tried it on the square "factory":
```csharp
public readonly struct Square : ISquareFactory<TQuantity, TSquare, TLinear>
  /* where snip */
{
    public Compound<TCreate, TQuantity, TLinear> Square => new( /* snip */ );
}
```
Which results in a naming conflict of the class name and the property:
```
'Square': member names cannot be the same as their enclosing type [quantities]csharp(CS0542)
```
So I went with SecondPower (inspired by some wikipedia research), as it resolves that conflict.

With the cubic factory I didn't have that problem, as I could resolve the naming conflict like so:

```csharp
public readonly struct Cube : ICubicFactory<TQuantity, TCubic, TLinear>
  /* where snip */
{
    public Compound<TCreate, TQuantity, TLinear> Cubic => new( /* snip */ );
}
```
where the property name must be the fluent part, i.e. `Square` & `Cubic` respectively.

But I now see that this all leads to confusion and I could have used "Quadratic" for the square factory....

Hmm, I'll need to rethink this...

Reply:
Ok, so I've committed a proposal:

- Rename higher order factories resembling the [naming for polynomials](https://en.wikipedia.org/wiki/Degree_of_a_polynomial#Names_of_polynomials_by_degree).
- The fluent api wins out over the naming based on polynomials
  - This is visible in the `Cube` factory. 
- Omit "Factory" suffixes  in all factory implementations
  - `SiFactory` -> `SoOnly`
  - which reads fluently in say `ElectricalCurrent` as `SiOnly<To, ElectricCurrent, IElectricCurrent>` 
  - I also want to reduce "clutter" in the factory namespace, i.e. not having to suffix everything with "Factory".

@RaduTerec what do you think?

Reply:
I just finished looking at the changes. I think it's fine to have the interface names containing 'factory' and not the implementation. After all the interface describes what the class will do.

Feel free to merge the PR! 😁